### PR TITLE
moved delay_u and m to BSP_PLL, also renamed with BSP prefix

### PR DESCRIPTION
**Pull request by ray4477**
# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [ ] Software feature has test associated with it. 
- [ ] Software feature has documentation for it updated.
- [ ] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [ ] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [ ] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

File: BSP/Inc/BSP_PLL.h [hunk 1]
```diff
@@ -19,4 +19,18 @@ void BSP_PLL_Init(void);
  */
 uint32_t BSP_PLL_GetSystemClock(void);
 
+/**
+ * @brief   Delays in param milli seconds
+ * @param   # of milliseconds to delay
+ * @return  None
+*/
+void BSP_delay_m(uint16_t milli);
```
Comment (manthanand) [inline BSP/Inc/BSP_PLL.h:27]: make this `BSP_PLL_delay_m`

Comment (manthanand) [reply BSP/Inc/BSP_PLL.h:27]: same for delay u

File: BSP/Simulator/Src/BSP_PLL.c [hunk 2]
```diff
@@ -33,3 +33,31 @@ uint32_t BSP_PLL_GetSystemClock(void) {
...
+
+/**
+ * @brief   Delays in param micro seconds
+ * @param   # of microseconds to delay
+ * @return  None
+*/
+void BSP_delay_u(uint16_t micro)
+{
+    uint32_t delay = BSP_PLL_GetSystemClock() / 1000000;
+	for(uint32_t i = 0; i < micro; i++)
```
Comment (manthanand) [inline BSP/Simulator/Src/BSP_PLL.c:45]: make these volatile for all for loops

**Comment by manthanand** [comment]
Make sure to change all calls to these functions in the test files as well.

**Review by manthanand — DISMISSED** [review]

File: Apps/Src/Temperature.c [hunk 3]
```diff
@@ -15,7 +15,7 @@
 // if you change this, you also need to change the calls to median()
 #define TEMPERATURE_MEDIAN_FILTER_DEPTH 3
 
-void delay_u(uint16_t micro);
+void BSP_PLL_delay_u(uint16_t micro);
```
Comment (manthanand) [inline Apps/Src/Temperature.c:18]: You don't need this here. Just need to include `BSP_PLL.h`

**Comment by manthanand** [comment]
please fill out qac and request more reviewers than just me.

File: BSP/Inc/BSP_PLL.h [hunk 4]
```diff
@@ -19,4 +19,18 @@ void BSP_PLL_Init(void);
  */
 uint32_t BSP_PLL_GetSystemClock(void);
 
+/**
+ * @brief   Delays in param milli seconds
+ * @param   # of milliseconds to delay
+ * @return  None
+*/
+void BSP_PLL_delay_m(uint16_t milli);
```
Comment (tiandahuang) [inline BSP/Inc/BSP_PLL.h:27]: change the ```uint16_t``` to ```uint32_t``` (and also for delay u)

File: BSP/Inc/BSP_PLL.h [hunk 5]
```diff
@@ -19,4 +19,18 @@ void BSP_PLL_Init(void);
...
+
+/**
+ * @brief   Delays in param micro seconds
+ * @param   # of microseconds to delay
+ * @return  None
+*/
+void BSP_PLL_delay_u(uint16_t micro);
```
Comment (tiandahuang) [inline BSP/Inc/BSP_PLL.h:34]: To keep consistent style, these functions should probably be named
```
BSP_PLL_DelayU()
```
and
```
BSP_PLL_DelayM()
```

File: BSP/Simulator/Src/BSP_PLL.c [hunk 6]
```diff
@@ -33,3 +33,31 @@ uint32_t BSP_PLL_GetSystemClock(void) {
     }
     return SystemCoreClock;
 }
+
+/**
+ * @brief   Delays in param micro seconds
+ * @param   # of microseconds to delay
+ * @return  None
+*/
+void BSP_PLL_delay_u(uint16_t micro)
```
Comment (tiandahuang) [inline BSP/Simulator/Src/BSP_PLL.c:42]: I'm not sure we need these delays for the simulator, I think having ```return``` would suffice

File: BSP/Simulator/Src/BSP_PLL.c [hunk 7]
```diff
@@ -33,3 +33,31 @@ uint32_t BSP_PLL_GetSystemClock(void) {
...
 }
+
+/**
+ * @brief   Delays in param micro seconds
+ * @param   # of microseconds to delay
+ * @return  None
+*/
+void BSP_PLL_DelayU(uint32_t micro)
+{
+    uint32_t delay = BSP_PLL_GetSystemClock() / 1000000;
```
Comment (tiandahuang) [inline BSP/Simulator/Src/BSP_PLL.c:44]: delay in sim too?

**Review by tiandahuang — APPROVED** [review]

File: BSP/Simulator/Src/BSP_PLL.c (hunk 7 above)
Comment (manthanand) [reply BSP/Simulator/Src/BSP_PLL.c:44]: I don't think it hurts. If we need it later we can use it. For now, that function doesn't even get called in the sim.

**Pull request merged**